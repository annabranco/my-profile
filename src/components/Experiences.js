import React from 'react';
import '../styles/components/Experiences.css';

class Experiences extends React.Component {

	constructor(props) {
		super(props)

		this.state = {
			opened: false,
			read: false
		}
	}

	open = () => {
		if (!this.state.opened) {

			this.setState({
				opened: true,
				read: true
			 });
			document.querySelector('.section__experiences').classList.add('opened');
			this.props.markAsRead();
		}
	}

	close = () => {
		this.setState({ opened: false });
		document.querySelector('.section__experiences').classList.remove('opened');
	}

	render () {

		const text = this.props.texts.Experiences;
		const language = this.props.language;

		return (
			<React.Fragment>

				{!this.props.viewedExperiences && !this.state.read ?
					<React.Fragment>
						<p className="seabed__findSomething">{text[language].find}</p>
						<p className="seabed__findSomething">{text[language].investigate}</p>
					</React.Fragment>
				: null
				}


				<section className="section__experiences" onClick={this.open}>
					{!this.state.opened ?
						<React.Fragment>
							<p className="experiences__text-fake">- - --- - --</p>
							<p className="experiences__text-fake">- -- -- - --</p>
							<p className="experiences__text-fake">- -- ---- --</p>
							<p className="experiences__text-fake">- - - --- --</p>
						</React.Fragment>
					:
					<div className="experiences__outer">
						<button className="seabed__click2close" onClick={this.close}>{this.props.texts.Seabed[language].click2close}</button>

						<div className="experiences__inner">
							<div className="experiences__inner--year">09/2017 - 05/2018</div>
							<div className="experiences__inner--year">10/2015 - 11/2015</div>
							<div className="experiences__inner--year">05/2005 - 12/2014</div>

						</div>
						<div className="experiences__inner">
							<div className="experiences__verticalBar"></div>
						</div>
						<div className="experiences__inner">
							<div className="experiences__inner--horizontalBar">
								<div className="experiences__horizontarBar-psy"></div>
								<h2 className="experiences__title">{text[language].ict}</h2>
								<p className="experiences__details">Servicios Profesionales Sociales, Madrid.
									<img className="experiences__flag" src="https://www.countryflags.io/es/flat/16.png" alt={text[language].spain} title={text[language].spain} />
								</p>
								<p className="experiences__details">{text[language].ictDetails}</p>
							</div>
							<div className="experiences__inner--horizontalBar">
								<div className="experiences__horizontarBar-ir"></div>
								<h2 className="experiences__title">{text[language].eru}</h2>
								<p className="experiences__details">Cruz Roja Española
									<img className="experiences__flag" src="https://www.countryflags.io/es/flat/16.png" alt={text[language].spain} title={text[language].spain}  />
								</p>
								<p className="experiences__details">{text[language].eruDetails}</p>

							</div>
							<div className="experiences__inner--horizontalBar">
								<div className="experiences__horizontarBar-master"></div>
								<h2 className="experiences__title">{text[language].tj}</h2>
								<p className="experiences__details">Tribunal de Justiça do Estado do Rio de Janeiro <img className="experiences__flag" src="https://www.countryflags.io/br/flat/16.png" alt={text[language].brazil} title={text[language].brazil}  />
								</p>
								<p className="experiences__details">{text[language].tjDetails}</p>

							</div>

						</div>
						<p className="experiences__more">{text[language].linkedin}<a className="experiences__social--link" href="https://www.linkedin.com/in/annabranco/" target="_Blank" rel="noopener noreferrer">
								<i className="fab fa-linkedin-in icon-linkedin"></i>
							</a>
						</p>
					</div>
					}
				</section>

			</React.Fragment>

		);
	}

}

export default Experiences;
