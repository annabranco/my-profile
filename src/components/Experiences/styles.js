import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import { Sea } from '../../assets/images';
import {
  colorBlueLight,
  fontTitle,
  colorRedDark,
  fontTitleAlt,
  colorBlueDark,
  fontSubtitle
} from '../../styles/theme';

export const City = styled.span`
  font-size: 0.8rem;
  font-style: italic;
`;
City.displayName = 'City';

export const Company = styled.p`
  margin: 8px 0 0 15px;
  font-family: ${fontSubtitle};
  font-size: 1rem;
`;
Company.displayName = 'Company';

export const CountryFlag = styled.img`
  margin: 0 5px;
  vertical-align: middle;
`;
CountryFlag.displayName = 'Country Flag';

export const DateArea = styled.div`
  width: 12%;
`;
DateArea.displayName = 'Date Area';

export const Details = styled.p`
  margin-top: 10px;
`;
Details.displayName = 'Details';

export const DetailsArea = styled.div`
  margin-left: 40px;
  transform: translate(800px, 0);
  transition: all 4s ease;

  ${({ visible }) =>
    visible &&
    css`
      width: 80%;
      transform: translate(0, 0);
    `}
`;
DetailsArea.displayName = 'Details Area';

export const ExperiencesArea = styled.div`
  position: relative;
  background-image: linear-gradient(
      rgba(230, 239, 251, 0.98),
      rgba(230, 239, 251, 0.93),
      rgba(189, 214, 246, 0.98)
    ),
    ${`url(${Sea})`};
  background-position: center;
  height: auto;
`;
ExperiencesArea.displayName = 'Experiences Area';

export const ExperiencesWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: flex-end;
  margin: 0 auto;
  padding-right: 5vw;
  height: auto;
  width: 100%;
`;
ExperiencesWrapper.displayName = 'Experiences Wrapper';

export const ExperienceItem = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  margin: 20px 0;
  height: auto;
  width: 100%;
  opacity: 0;
  transition: opacity ease 3s;

  ${({ visible }) =>
    visible &&
    css`
      opacity: 1;
    `}
`;
ExperienceItem.displayName = 'Experience Item';

export const TextDate = styled.p`
  font-family: ${fontTitle};
  font-size: 1.2em;
  color: ${colorRedDark};
  text-align: center;
`;
TextDate.displayName = 'TextDate';

export const Title = styled.h3`
  display: inline;
  margin-bottom: 20px;
  font-family: ${fontTitleAlt};
  font-size: 1.4em;
  color: ${colorBlueDark};
`;
Title.displayName = 'Title';

export const VerticalBar = styled.div`
  position: absolute;
  left: 10%;
  border-right: 5px solid ${rgba(colorBlueLight, 0.3)};
  height: 100%;
  width: 0;
`;
VerticalBar.displayName = 'Vertical Bar';
