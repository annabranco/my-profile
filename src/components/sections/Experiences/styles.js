import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import { Sea } from '../../../assets/images';
import {
  colorBlueLight,
  fontTitle,
  colorRedDark,
  fontTitleAlt,
  colorBlueDark,
  fontSubtitle
} from '../../../styles/theme';

export const City = styled.span`
  font-size: 0.7rem;
  font-style: italic;

  @media all and (min-width: 768px) {
    font-size: 0.8rem;
  }
`;
City.displayName = 'City';

export const Company = styled.p`
  margin: 8px 0 0 15px;
  font-family: ${fontSubtitle};
  font-size: 0.8rem;

  @media all and (min-width: 768px) {
    font-size: 1rem;
  }
`;
Company.displayName = 'Company';

export const CountryFlag = styled.img`
  margin: 0 5px;
  vertical-align: middle;
`;
CountryFlag.displayName = 'Country Flag';

export const DateArea = styled.div`
  width: 12%;
`;
DateArea.displayName = 'Date Area';

export const Details = styled.p`
  margin-top: 10px;
`;
Details.displayName = 'Details';

export const DetailsArea = styled.div`
  margin-top: 10px;
  margin-left: 40px;
  transform: translate(800px, 0);
  transition: all 4s ease;

  ${({ visible }) =>
    visible &&
    css`
      width: 80%;
      transform: translate(0, 0);
    `}
  @media all and (min-width: 768px) {
    margin-top: 0;
  }

  ${({ modal }) =>
    modal &&
    css`
      background: rgba(255, 255, 255, 0.8);
      padding: 20px;
      border-radius: 5px;
    `}
`;
DetailsArea.displayName = 'Details Area';

export const ExperiencesArea = styled.div`
  position: relative;
  background-image: linear-gradient(
      rgba(230, 239, 251, 0.98),
      rgba(230, 239, 251, 0.93),
      rgba(189, 214, 246, 0.98)
    ),
    ${`url(${Sea})`};
  background-position: center;
  height: ${`${window.innerHeight}px`};

  @media all and (min-width: 768px) {
    height: auto;
  }
`;
ExperiencesArea.displayName = 'Experiences Area';

export const ExperiencesWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: flex-end;
  margin: 130px auto 0;
  padding-right: 5vw;
  height: 50%;

  @media all and (min-width: 768px) {
    margin: 0 auto;
    height: auto;
  }
`;
ExperiencesWrapper.displayName = 'Experiences Wrapper';

export const ExperienceItem = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  margin: 10px 0;
  height: auto;
  width: 100%;
  opacity: 0;
  transition: opacity ease 3s;

  ${({ visible }) =>
    visible &&
    css`
      opacity: 1;
    `}

  @media all and (min-width: 768px) {
    margin: 20px 0;
  }
`;
ExperienceItem.displayName = 'Experience Item';

export const TextDate = styled.p`
  font-family: ${fontTitle};
  font-size: 0.8em;
  color: ${colorRedDark};
  text-align: center;

  @media all and (min-width: 768px) {
    font-size: 1.2em;
  }
`;
TextDate.displayName = 'TextDate';

export const Title = styled.h3`
  display: inline;
  margin-bottom: 20px;
  font-family: ${fontTitleAlt};
  font-size: 1em;
  color: ${colorBlueDark};

  @media all and (min-width: 768px) {
    font-size: 1.4em;
  }
`;
Title.displayName = 'Title';

export const VerticalBar = styled.div`
  position: absolute;
  left: 10%;
  border-right: 5px solid ${rgba(colorBlueLight, 0.3)};
  height: 100%;
  width: 0;
`;
VerticalBar.displayName = 'Vertical Bar';
