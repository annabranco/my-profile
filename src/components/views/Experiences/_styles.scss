.section__experiences {
	z-index          : 3;
	position         : absolute;
	bottom           : 50px;
	left             : 20%;
	opacity          : 0.5;
	transform        : perspective(5em) rotateX(50deg);
	outline          : none;
	border           : 1px solid $black;
	border-radius    : 10px 0;
	background-image : linear-gradient(to top right, $white,#e6f4f4,$blue--light);
	height           : 40px;
	width            : 60px;
	box-sizing       : border-box;
	overflow          : hidden;
	cursor           : pointer;
	box-shadow       : 0 4px 5px 0 rgba($black, 0.4),
										 inset 0 1px 5px 1px rgba($white, 0.7);

	@media all and (min-width: 768px) {
		left : 40%;
	}

	&:hover {
		opacity : 0.7;
		border  : 3px solid $yellow--bright;
	}

	&.show {

		top           : 40px;
		left          : 0;
		opacity       : 0.9;
		transform     : none;
		border        : 2px solid $black;
		border-radius : 40px 0;
		height        : 90vh;
		width         : 100vw;
		padding       : 5px;
		cursor        : default;

		@media all and (min-width: 768px) {
			left    : 20vw;
			height  : 90vh;
			width   : 75vw;
			padding : 15px;
		}

		&:hover {
			border : 2px solid $black;
		}
	}
}

.experiences__text-fake {
	width       : 100%;
	font-size   : 0.9rem;
	line-height : 0.8;
	text-align  : center;
}

.experiences__outer {
	display         : flex;
	justify-content : flex-start;
	align-items     : flex-end;
	margin          : 0 auto;
	height          : 100%;
	width           : 100%;
}

.experiences__inner {
	height : 100%;

	&:last-of-type {
		width : 100%;
	}
}

.experiences__inner--year {
	margin-right : 5px;
	width        : 40px;
	font-family  : $font__title;
	font-size    : 0.8em;
	color        : $red--dark;
	text-align   : center;

	@media all and (min-width: 321px) {
		margin-right : 10px;
		width        : 60px;
		font-size    : 1.2em;
	}

	@media all and (min-width: 768px) {
		margin-right : 15px;
		width        : 60px;
	}
}

.experiences__verticalBar {
	border-right : 5px solid rgba($blue--light, 0.3);
	height       : 80vh;

	@media all and (min-width: 768px) {
		height : 85vh;
	}
}

.experiences__inner--horizontalBar,
.experiences__inner--year {
	margin-top : 50px;
	height     : 25%;

	@media all and (min-width: 321px) {
		margin-top : 25px;

		&:first-of-type {
			margin-top : 60px;
		}

		&:last-of-type {
			margin-top : 10px;
		}
	}

	@media all and (min-width: 768px) {
		margin-top : 40px;
	}

	&:first-of-type {
		margin-top : 35px;
	}

	&:last-of-type {
		margin-top : 10px;
	}
}

.experiences__horizontarBar-adalab,
.experiences__horizontarBar-ir,
.experiences__horizontarBar-master,
.experiences__horizontarBar-psy {
	display    : inline-block;
	border-top : 5px solid rgba($blue--light, 0.3);
}

.experiences__horizontarBar-psy {
	width : 10%;
}

.experiences__horizontarBar-ir {
	width : 10%;

	@media all and (min-width: 768px) {
		width : 30%;
	}
}

.experiences__horizontarBar-master {
	width : 10%;

	@media all and (min-width: 768px) {
		width : 30%;
	}
}

.experiences__title {
	display     : inline;
	margin-left : 20px;
	font-family : $font__title-alt;
	font-size   : 1rem;
	color       : $blue--dark;

	@media all and (min-width: 321px) {
		font-size : 1.4rem;
	}

	&:hover {
		letter-spacing : 0.1rem;
		color          : $yellow--dark;
	}
}

.experiences__details {
	margin      : 5px 10px;
	font-family : $font__subtitle;
	font-size   : 0.9rem;
	line-height : 1;

	@media all and (min-width: 768px) {
		font-size   : 1rem;
		line-height : 2;
	}
}

.experiences__flag {
	margin-left    : 5px;
	vertical-align : middle;
}

.experiences__more {
	position  : absolute;
	bottom    : 5px;
	right     : 35px;
	font-size : 0.8rem;

	@media all and (min-width: 321px) {
		bottom    : 15px;
		font-size : 1rem;
	}
}

.experiences__social--link {
	margin-left    : 5px;
	vertical-align : super;
}
