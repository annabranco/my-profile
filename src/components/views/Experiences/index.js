import React from 'react';
import { PropTypes } from 'prop-types';
import { getFlagURL } from '../../../utils/icons';
import { experiencesPropType, globalTextsPropType } from '../../../types';
import {
  SectionExperiences,
  VerticalBar,
  ExperiencesWrapper,
  ExperienceItem,
  DateField,
  TextDate,
  DetailsField,
  ExperienceTitle,
  ExperienceCompany,
  CountryFlag,
  ExperienceDetails,
  ExperiencePlace
} from './styles';

const EXPERIENCE_ON_TOP = 'newer'; // newer or older

const Experiences = ({ texts, experiences, language, cuePointsActivated }) => {
  const customOrder = (x, y) => {
    if (EXPERIENCE_ON_TOP === 'older') {
      return x.dateBeginValue - y.dateBeginValue;
    }
    return y.dateBeginValue - x.dateBeginValue;
  };

  return (
    <SectionExperiences id="experiences">
      <VerticalBar />
      <ExperiencesWrapper>
        {experiences.sort(customOrder).map(
          item =>
            item.visible && (
              <ExperienceItem
                key={item.dateBeginValue}
                visible={cuePointsActivated.includes(item.id)}
              >
                <DateField>
                  {!item.dateEnd && <TextDate>{texts.since}</TextDate>}
                  <TextDate>{item.dateBegin}</TextDate>
                  {item.dateEnd && (
                    <>
                      <TextDate>-</TextDate>
                      <TextDate>{item.dateEnd}</TextDate>
                    </>
                  )}
                </DateField>
                <DetailsField>
                  <ExperienceTitle>{item.title[language]}</ExperienceTitle>
                  <ExperienceCompany>
                    {item.company}
                    <CountryFlag
                      src={getFlagURL(
                        item.country.countryCode,
                        'flat',
                        'small'
                      )}
                      alt={item.country[language]}
                    />
                    <ExperiencePlace>{item.place}</ExperiencePlace>
                  </ExperienceCompany>
                  <ExperienceDetails>
                    {item.details[language]}
                  </ExperienceDetails>
                </DetailsField>
              </ExperienceItem>
            )
        )}
      </ExperiencesWrapper>
    </SectionExperiences>
  );
};

Experiences.propTypes = {
  experiences: PropTypes.arrayOf(experiencesPropType).isRequired,
  language: PropTypes.string.isRequired,
  texts: globalTextsPropType.isRequired,
  cuePointsActivated: PropTypes.arrayOf(PropTypes.string).isRequired
};

export default Experiences;
