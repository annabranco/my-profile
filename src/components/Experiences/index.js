import React from 'react';
import { useSelector } from 'react-redux';
import { instanceOf } from 'prop-types';
import {
  currentLanguageSelector,
  experiencesSelector,
  globalTextsSelector
} from '../../redux/selectors';
import { getFlagURL } from '../../utils/icons';
import {
  City,
  Company,
  CountryFlag,
  DateArea,
  Details,
  DetailsArea,
  ExperienceItem,
  ExperiencesArea,
  ExperiencesWrapper,
  TextDate,
  Title,
  VerticalBar
} from './styles';
import { HorizontalBar } from '../elements/HorizontalBar/styles';

const EXPERIENCE_ON_TOP = 'newer'; // newer or older

const Experiences = ({ cuePointsActivated }) => {
  const experiences = useSelector(experiencesSelector);
  const languageSelected = useSelector(currentLanguageSelector);
  const texts = useSelector(globalTextsSelector);

  const customOrder = (x, y) => {
    if (EXPERIENCE_ON_TOP === 'older') {
      return x.dateBeginValue - y.dateBeginValue;
    }
    return y.dateBeginValue - x.dateBeginValue;
  };

  return (
    <ExperiencesArea>
      <VerticalBar />
      <ExperiencesWrapper>
        {experiences.sort(customOrder).map(
          item =>
            item.visible && (
              <ExperienceItem
                key={item.dateBeginValue}
                visible={cuePointsActivated.has(item.id)}
              >
                <HorizontalBar
                  border="2px"
                  margin="9%"
                  moveY="25px"
                  width="20%"
                />
                <DateArea>
                  {!item.dateEnd && <TextDate>{texts.since}</TextDate>}
                  <TextDate>{item.dateBegin}</TextDate>
                  {item.dateEnd && (
                    <>
                      <TextDate>-</TextDate>
                      <TextDate>{item.dateEnd}</TextDate>
                    </>
                  )}
                </DateArea>
                <DetailsArea visible={cuePointsActivated.has(item.id)}>
                  <Title>{item.title[languageSelected]}</Title>
                  <Company>
                    {item.company}
                    <CountryFlag
                      src={getFlagURL(
                        item.country.countryCode,
                        'flat',
                        'small'
                      )}
                      alt={item.country[languageSelected]}
                    />
                    <City>{item.place}</City>
                  </Company>
                  <Details>{item.details[languageSelected]}</Details>
                </DetailsArea>
              </ExperienceItem>
            )
        )}
      </ExperiencesWrapper>
    </ExperiencesArea>
  );
};

Experiences.propTypes = {
  cuePointsActivated: instanceOf(Set).isRequired
};

export default Experiences;
