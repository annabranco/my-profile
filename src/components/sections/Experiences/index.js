import React from 'react';
import { useSelector } from 'react-redux';
import { instanceOf } from 'prop-types';
import {
  currentLanguageSelector,
  experiencesSelector,
  globalTextsSelector
} from '../../../redux/selectors';
import { getFlagURL } from '../../../utils/icons';
import {
  City,
  Company,
  CountryFlag,
  DateArea,
  Details,
  DetailsArea,
  ExperienceItem,
  ExperiencesArea,
  ExperiencesWrapper,
  TextDate,
  Title,
  VerticalBar
} from './styles';
import { HorizontalBar } from '../../elements/HorizontalBar/styles';
import { isDesktop } from '../../../utils/device';
import AppModal from '../../mainComponents/AppModal';
import { useStateWithLabel } from '../../../utils/hooks';

const EXPERIENCE_ON_TOP = 'newer'; // newer or older

const Experiences = ({ cuePointsActivated }) => {
  const experiences = useSelector(experiencesSelector);
  const languageSelected = useSelector(currentLanguageSelector);
  const texts = useSelector(globalTextsSelector);

  const [selectedExperience, selectExperience] = useStateWithLabel(
    null,
    'selectedExperience'
  );

  const customOrder = (x, y) => {
    if (EXPERIENCE_ON_TOP === 'older') {
      return x.dateBeginValue - y.dateBeginValue;
    }
    return y.dateBeginValue - x.dateBeginValue;
  };

  return (
    <ExperiencesArea>
      <VerticalBar />
      <ExperiencesWrapper>
        {experiences.sort(customOrder).map(
          (item, index) =>
            item.visible && (
              <ExperienceItem
                key={item.dateBeginValue}
                onClick={isDesktop ? () => null : () => selectExperience(item)}
                visible={cuePointsActivated.has(item.id)}
              >
                <HorizontalBar
                  border="2px"
                  margin="9%"
                  moveY="25px"
                  formationItems={experiences.length}
                  index={index}
                />
                <DateArea>
                  {!item.dateEnd && <TextDate>{texts.since}</TextDate>}
                  <TextDate>{item.dateBegin}</TextDate>
                  {item.dateEnd && (
                    <>
                      <TextDate>-</TextDate>
                      <TextDate>{item.dateEnd}</TextDate>
                    </>
                  )}
                </DateArea>
                <DetailsArea visible={cuePointsActivated.has(item.id)}>
                  <Title>{item.title[languageSelected]}</Title>
                  <Company>
                    {item.company}
                    <CountryFlag
                      src={getFlagURL(
                        item.country.countryCode,
                        'flat',
                        'small'
                      )}
                      alt={item.country[languageSelected]}
                    />
                    <City>{item.place}</City>
                  </Company>
                  {isDesktop && (
                    <Details>{item.details[languageSelected]}</Details>
                  )}
                </DetailsArea>
              </ExperienceItem>
            )
        )}
      </ExperiencesWrapper>
      {selectedExperience && (
        <AppModal closeAction={() => selectExperience(null)}>
          <DetailsArea
            visible={cuePointsActivated.has(selectedExperience.id)}
            modal
          >
            <Title>{selectedExperience.title[languageSelected]}</Title>
            <Company>
              {selectedExperience.company}
              <CountryFlag
                src={getFlagURL(
                  selectedExperience.country.countryCode,
                  'flat',
                  'small'
                )}
                alt={selectedExperience.country[languageSelected]}
              />
              <City>{selectedExperience.place}</City>
            </Company>
            <Details>{selectedExperience.details[languageSelected]}</Details>
          </DetailsArea>
        </AppModal>
      )}
    </ExperiencesArea>
  );
};

Experiences.propTypes = {
  cuePointsActivated: instanceOf(Set).isRequired
};

export default Experiences;
