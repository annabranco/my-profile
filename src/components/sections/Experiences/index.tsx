import { ReactElement } from 'react';
import { useSelector } from 'react-redux';
import { instanceOf } from 'prop-types';
import {
  IExperiencesType,
  IGlobalTexts,
  LanguageCode
} from '../../../types/interfaces';
import {
  currentLanguageSelector,
  experiencesSelector,
  globalTextsSelector
} from '../../../redux/selectors';
import { getFlagURL } from '../../../utils/icons';
import { isDesktop } from '../../../utils/device';
import { HorizontalBar } from '../../elements/HorizontalBar/styles';
import AppModal from '../../mainComponents/AppModal';
import { useStateWithLabel } from '../../../utils/hooks';
import {
  City,
  Company,
  CountryFlag,
  DateArea,
  Details,
  DetailsArea,
  ExperienceItem,
  ExperiencesArea,
  ExperiencesWrapper,
  TextDate,
  Title,
  VerticalBar
} from './styles';

const EXPERIENCE_ON_TOP = 'newer'; // newer or older

interface Props {
  cuePointsActivated: Set<string>;
}

const Experiences = ({ cuePointsActivated }: Props): ReactElement => {
  const experiences: IExperiencesType[] = useSelector(experiencesSelector);
  const languageSelected: LanguageCode = useSelector(currentLanguageSelector);
  const texts: IGlobalTexts = useSelector(globalTextsSelector);

  const [selectedExperience, selectExperience] = useStateWithLabel<
    IExperiencesType | Record<string, never>
  >({}, 'selectedExperience');

  const customOrder = (x: IExperiencesType, y: IExperiencesType) => {
    if (EXPERIENCE_ON_TOP === 'newer') {
      return y.dateBeginValue - x.dateBeginValue;
    }
    return x.dateBeginValue - y.dateBeginValue;
  };

  return (
    <ExperiencesArea>
      <VerticalBar />
      <ExperiencesWrapper>
        {experiences.sort(customOrder).map(
          (item, index) =>
            item.visible && (
              <ExperienceItem
                key={item.dateBeginValue}
                onClick={isDesktop ? () => null : () => selectExperience(item)}
                visible={cuePointsActivated.has(item.id)}
              >
                <HorizontalBar
                  border="2px"
                  formationItems={experiences.length}
                  index={index}
                  margin="9%"
                  moveY="25px"
                />
                <DateArea>
                  {!item.dateEnd && <TextDate>{texts.since}</TextDate>}
                  <TextDate>{item.dateBegin}</TextDate>
                  {item.dateEnd && (
                    <>
                      <TextDate>-</TextDate>
                      <TextDate>{item.dateEnd}</TextDate>
                    </>
                  )}
                </DateArea>
                <DetailsArea visible={cuePointsActivated.has(item.id)}>
                  <Title>{item.title[languageSelected]}</Title>
                  <Company>
                    {item.company}
                    <CountryFlag
                      alt={item.country[languageSelected]}
                      src={getFlagURL({
                        country: item.country.countryCode,
                        style: 'round3d',
                        size: 'small'
                      })}
                    />
                    <City>{item.place}</City>
                  </Company>
                  {isDesktop && (
                    <Details>{item.details[languageSelected]}</Details>
                  )}
                </DetailsArea>
              </ExperienceItem>
            )
        )}
      </ExperiencesWrapper>
      {selectedExperience.title && (
        <AppModal closeAction={() => selectExperience({})}>
          <DetailsArea
            visible={cuePointsActivated.has(selectedExperience.id)}
            modal
          >
            <Title>{selectedExperience.title[languageSelected]}</Title>
            <Company>
              {selectedExperience.company}
              <CountryFlag
                src={getFlagURL({
                  country: selectedExperience.country.countryCode,
                  size: 'small',
                  style: 'flat3d'
                })}
                alt={selectedExperience.country[languageSelected]}
              />
              <City>{selectedExperience.place}</City>
            </Company>
            <Details>{selectedExperience.details[languageSelected]}</Details>
          </DetailsArea>
        </AppModal>
      )}
    </ExperiencesArea>
  );
};

Experiences.propTypes = {
  cuePointsActivated: instanceOf(Set).isRequired
};

export default Experiences;
