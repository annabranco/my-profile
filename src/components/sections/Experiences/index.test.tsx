import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import Experiences from '.';
import { setupMockProvider } from '../../../tests/mocks';

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

describe('< Experiences >', () => {
  let MockProvider;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    wrapper = shallow(
      <MockProvider>
        <Experiences cuePointsActivated={new Set()} />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  xit('should render all experiences received from server', () => {});

  xit('should dislay the experiences correctly ordered according to the custom order configured', () => {});

  xit('should display date, title and details of each experience', () => {});

  xit('should display the country flag of each experience', () => {});

  xit('should open the details on a modal when an experience is clicked on devices', () => {});

  xit('should display animation when cuepoints are activated', () => {});

  xit('should translate all texts when the language is changed', () => {});
});
