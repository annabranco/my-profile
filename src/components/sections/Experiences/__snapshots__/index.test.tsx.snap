// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Experiences > matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Experiences
    cuePointsActivated={Set {}}
  />
</Provider>
`;
